src/test/java/edu/cornell/kfs/fp/document/service/impl/CuDisbursementVoucherDefaultDueDateServiceImplTest.java

```diff
@@ -92,37 +100,57 @@ public void testFindDefaultDueDateBadParamEmptyNull() {
         }
         assertTrue("This should have generated an IllegalStateException", caughtError);
     }
-    
-    protected Calendar buildCalendar(int year, int month, int day) {
-        Calendar calendar = Calendar.getInstance();
-        calendar.set(Calendar.YEAR, year);
-        calendar.set(Calendar.MONTH, month);
-        calendar.set(Calendar.DAY_OF_MONTH, day);
-        return calendar;
-    }
-    
-    protected void assertServiceDateEqualsExpectedDate(Date expectedDate) {
+
+    protected void assertServiceDateEqualsExpectedDate(LocalDate expectedDate) {
         Date actualDate = cuDisbursementVoucherDefaultDueDateServiceImpl.findDefaultDueDate();
-        assertEquals(dateFormat.format(expectedDate), dateFormat.format(actualDate));
+        assertEquals(java.sql.Date.valueOf(expectedDate), actualDate);
     }
-    
-    protected void prepCuDisbursementVoucherDefaultDueDateServiceImpl(String numberOfDays, Calendar currentCalendar) {
-        cuDisbursementVoucherDefaultDueDateServiceImpl.setDateTimeService(buildMockDateTimeService(currentCalendar));
+
+    protected void prepCuDisbursementVoucherDefaultDueDateServiceImpl(String numberOfDays) {
         cuDisbursementVoucherDefaultDueDateServiceImpl.setParameterService(buildMockParameterService(numberOfDays));
     }
-    
+
     protected ParameterService buildMockParameterService(String numberOfDays) {
         ParameterService service = Mockito.mock(ParameterService.class);
-        Mockito.when(service.getParameterValueAsString(DisbursementVoucherDocument.class, 
-                CuFPParameterConstants.DisbursementVoucherDocument.NUMBER_OF_DAYS_FOR_DEFAULT_DV_DUE_DATE)).thenReturn(numberOfDays);
+        Mockito.when(service.getParameterValueAsString(DisbursementVoucherDocument.class,
+                CuFPParameterConstants.DisbursementVoucherDocument.NUMBER_OF_DAYS_FOR_DEFAULT_DV_DUE_DATE))
+                .thenReturn(numberOfDays);
         return service;
     }
-    
+
     protected DateTimeService buildMockDateTimeService(Calendar currentCalendar) {
         DateTimeService service = Mockito.mock(DateTimeService.class);
         Mockito.when(service.getCurrentCalendar()).thenReturn(currentCalendar);
         return service;
-        
+
+    }
+
+    private java.sql.Date getSqlDate(InvocationOnMock invocation) {
+        java.time.LocalDate p = invocation.getArgument(0);
+        return java.sql.Date.valueOf(p);
+    }
+
+    private DateTimeService buildSpiedDateTimeService(TestDateTimeServiceImpl actualDateTimeService, int year,
+            Month month, int dayOfMonth) {
+        TestDateTimeServiceImpl dateTimeService = Mockito.spy(actualDateTimeService);
+        Mockito.doAnswer(invocation -> getMockLocalDateNow(year, month, dayOfMonth)).when(dateTimeService)
+                .getLocalDateNow();
+        Mockito.doAnswer(invocation -> getSqlDate(invocation)).when(dateTimeService).getSqlDate(any(LocalDate.class));
```

Comment:
Does the `getSqlDate()` method need to be mocked? From what I can tell, the underlying service should be able to handle those method calls successfully.